File: Cpp/896_Monotonic_Array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        bool increasing = false;
        bool decreasing = false;
        for(int i=1; i<A.size(); ++i){
            if(A[i]-A[i-1] == 0) continue;
            else if(A[i]-A[i-1]>0) increasing = true;
            else decreasing = true;
            
            if(increasing && decreasing) return false;
        }
        
        return true;
    }
};

File: Kotlin/896_Monotonic_Array.kt
class Solution {
    fun isMonotonic(nums: IntArray): Boolean {
        if (nums.size < 3) return true
        
        val ascending = nums[0] < nums[nums.lastIndex]
        for(i in 1..nums.lastIndex) {
            if((ascending && nums[i] < nums[i - 1])
               || (!ascending && nums[i] > nums[i - 1])) return false
        }
        
        return true
    }
}